src/main/java/domain/Name.java

```diff
@@ -0,0 +1,47 @@
+package domain;
+
+import java.util.Objects;
+
+class Name {
+    static final int MAX_NAME_LENGTH = 5;
+    private final String name;
+
+    Name(String name) {
+        validateNameLength(name);
+        validateNameCharacters(name);
+        this.name = name;
+    }
+
+    private static void validateNameCharacters(String name) {
+        if (name.matches("^[^a-zA-Z]+$")) {
```

Comment:
String 의 `matches` 메서드는 Pattern 객체를 내부적으로 계속 새로 생성하는데요

Pattern 객체는 생성 비용이 비싼 객체중 하나 입니다 😃 
Pattern 객체를 재사용 할 수 있는 방법은 없을까요?

String의 matches 메서드 내부를 살펴볼까요?
matches 메서드 내부에서는 Pattern 객체의 matches 메서드를 호출하고 있어요

다음은 Pattern 클래스의 matches 메서드에 적힌 주석입니다.
```java
Compiles the given regular expression and attempts to match the given input against it.
An invocation of this convenience method of the form
  Pattern.matches(regex, input);
behaves in exactly the same way as the expression
  Pattern.compile(regex).matcher(input).matches()
If a pattern is to be used multiple times, compiling it once and reusing it will be more efficient than invoking this method each time.
```